aptos-move/framework/supra-framework/sources/pbo_delegation_pool.move

```diff
@@ -1601,10 +1586,237 @@ module supra_framework::pbo_delegation_pool {
+                error::permission_denied(ENOT_AUTHORIZED)
+            );
+        };
+        assert!(amount > 0, error::invalid_argument(EWITHDRAW_ZERO_STAKE));
+        withdraw_internal(
+            borrow_global_mut<DelegationPool>(pool_address),
+            delegator_address,
+            amount,
+            signer::address_of(multisig_admin),
+        );
+    }
+
+    /// For each `delegator` in `delegators`, locks up to the amount of stake specified in the same index of
+    /// `max_stakes_to_lock`, which represents the upper bound on the total amount of locked stake that the
+    /// delegator should have when the function terminates. The locked amount is subject to the vesting
+    /// schedule specified when the delegation pool corresponding to `pool_address` was created.
+    /// 
+    /// Note that this function is only temporarily intended to work as specified above and exists to enable The
+    /// Supra Foundation to ensure that the allocations of all investors are subject to the terms specified in the
+    /// corresponding legal contracts. It will be deactivated before the validator set it opened up to external 
+    /// validator-owners to prevent it from being abused.
+    public entry fun lock_delegators_stakes(
+        multisig_admin: &signer,
+        pool_address: address,
+        delegators: vector<address>,
+        max_stakes_to_lock: vector<u64>
+    ) acquires DelegationPool, GovernanceRecords, BeneficiaryForOperator, NextCommissionPercentage {
+        // Ensure that the caller is the admin of the delegation pool.
+        {
+            assert!(
+                is_admin(signer::address_of(multisig_admin), pool_address),
+                error::permission_denied(ENOT_AUTHORIZED)
+            );
+        };
+
+        // Synchronize the delegation and stake pools before any user operation.
+        synchronize_delegation_pool(pool_address);
+
+        // Ensure that each `delegator` has enough active stake to cover the corresponding `stake_to_lock`.
+        let new_amounts_to_lock = vector::zip_map(
+            delegators,
+            max_stakes_to_lock,
+            |delegator, max_stake_to_lock| {
+                let (active, inactive, pending_inactive) = get_stake(pool_address, delegator);
+                let already_locked = 0;
+                let pool: &mut DelegationPool = borrow_global_mut<DelegationPool>(pool_address);
+
+                // See if the delegator already has a locked allocation.
+                if (table::contains(&pool.principle_stake, delegator)) {
+                    already_locked = *table::borrow(&pool.principle_stake, delegator);
+                };
+
+                if (max_stake_to_lock > already_locked) {
+                    let max_new_stake_to_lock = max_stake_to_lock - already_locked;
+                    // All `already_locked` stake must be `active`.
+                    let lockable_active_stake = active - already_locked;
```

Comment:
The assumption that `already_locked` stake must be `active` does not hold true. The amount in `principle_stake` does not change, and it only indicates the principle amount. The actual amount may differ depending upon `cumulative_unlocked_fraction`.

I believe it would be simpler to not worry about what is already locked and this method should target locking `max_stake_to_lock` (and not `max_new_stake_to_lock`)

Reply:
Thanks. I've updated the function.